src/components/CohortBuilder/QueriesResolver.js

```diff
@@ -32,28 +32,33 @@ class QueriesResolver extends React.Component {
     }
   }
 
-  update = () =>
-    this.taskQueue.enqueue(
-      () =>
-        new Promise(async resolve => {
-          const { queries = [], useCache = true } = this.props;
-          const body = JSON.stringify(
-            queries.map(q => ({
-              query: typeof q.query === 'string' ? q.query : print(q.query),
-              variables: q.variables,
-            })),
-          );
-          try {
-            if (!useCache) {
-              this.setState({ isLoading: true });
-            }
-            const data = useCache ? await this.cachedFetchData(body) : await this.fetchData(body);
-            this.setState({ data: data, isLoading: false }, resolve);
-          } catch (err) {
-            this.setState({ isLoading: false, error: err }, resolve);
-          }
-        }),
-    );
+  update = () => {
+    const { queries = [], useCache = true } = this.props;
+    return queries.length !== 0
+      ? this.taskQueue.enqueue(
+          () =>
+            new Promise(async resolve => {
+              const body = JSON.stringify(
+                queries.map(q => ({
+                  query: typeof q.query === 'string' ? q.query : print(q.query),
+                  variables: q.variables,
+                })),
+              );
+              try {
+                if (!useCache) {
```

Comment:
This was my hack, but `useCache` shouldn't be used here really, let's use its own `useLoadingState` prop based on our earlier chat?

Reply:
@hlminh2000 
So I ended up only needing to check for parent loading state for the Cards and not having to add a new loading state to QueriesResolver, so useCache is just the same as before. Only change to QueriesResolver is the no query on empty array. Do you want me to just take out the useCache setstate?